cmd/ipfs/util/signal.go

```diff
@@ -36,7 +42,16 @@ func (ih *IntrHandler) Close() error {
 // handler's wait group to ensure clean shutdown when Close() is called.
 func (ih *IntrHandler) Handle(handler func(count int, ih *IntrHandler), sigs ...os.Signal) {
 	notify := make(chan os.Signal, 1)
+
+	// register with Golang's stdlib
 	signal.Notify(notify, sigs...)
+
+	// if we have platform specific handler logic
+	// call it so it can inject signals from the OS
+	if nativeNotifier != nil {
+		nativeNotifier(ih, notify, sigs...)
+	}
```

Comment:
This should probably go _below_ the `ih.wg.Add` call (or maybe just move that goroutine to the top of the function?). Otherwise, we could:

1. Receive a window close event before we call `ih.wg.Add`
2. Call `ih.wg.Wait()`
3. Succeed.
4. Either the process gets killed _or_ we then call `ih.wg.Add(1)` and panic (because we've already called wait on the waitgroup.